### PR TITLE
Fix rating

### PR DESCRIPTION
**Pull request by DimitarShemet**

### CODE REVIEW

File: src/components/VideoRating.tsx [hunk 1]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
 interface VideoRatingProps {
   ratingProp?: number;
   votesCount?: number;
+  onGetNewRating: (rating: number) => void;
+  isAuthorized: boolean;
+  isRatingFreezen: boolean;
```
Comment (Terapool) [inline src/components/VideoRating.tsx:11]: isRatingFrozen

File: src/components/VideoRating.tsx [hunk 2]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
...
+  ratingProp,
+  votesCount,
+  onGetNewRating,
+  isAuthorized,
+  isRatingFreezen,
+}: VideoRatingProps) => {
   const [rating, setRating] = useState(0);
-  const [hover] = useState(0);
-
+  const [hover]=useState(0);
```
Comment (Terapool) [inline src/components/VideoRating.tsx:21]: Please, spaces: [hover] = ...

File: src/components/VideoRating.tsx [hunk 3]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
...
+  
   return (
     <div className="star-rating">
       {[...Array(5)].map((star, index) => {
         const ratingUnit: number = index + 1;
+
```
Comment (Terapool) [inline src/components/VideoRating.tsx:27]: const isRatingEnabled = isAuthorized && !isRatingFrozen

File: src/components/VideoRating.tsx [hunk 4]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
...
         return (
           <button
+            style={isAuthorized && !isRatingFreezen? { cursor: "pointer" } : { cursor: "auto" }}
```
Comment (Terapool) [inline src/components/VideoRating.tsx:30]: style={{cursor: isRatingEnabled ? "pointer" : "default"}}

but it is better here use className for toggling cursor

File: src/components/VideoRating.tsx [hunk 5]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
...
+  isRatingFreezen,
+}: VideoRatingProps) => {
   const [rating, setRating] = useState(0);
-  const [hover] = useState(0);
-
+  const [hover]=useState(0);
+  
   return (
     <div className="star-rating">
       {[...Array(5)].map((star, index) => {
```
Comment (Terapool) [inline src/components/VideoRating.tsx:25]: please, create const STARS_NUMBER = 5 and then use it ...Array(STARS_NUMBER)

File: src/components/VideoRating.tsx [hunk 6]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
...
         const ratingUnit: number = index + 1;
+
         return (
           <button
+            style={isAuthorized && !isRatingFreezen? { cursor: "pointer" } : { cursor: "auto" }}
             type="button"
             key={ratingUnit}
             className={ratingUnit <= (hover || rating) ? "on" : "off"}
```
Comment (Terapool) [inline src/components/VideoRating.tsx:33]: use classNames library for complex logic

File: src/pages/VideoViewPage.tsx [hunk 7]
```diff
@@ -24,17 +27,27 @@ interface CategoriesResponce {
...
   const [lessonsArr, setLessonsArr] = useState<Lesson[]>([]);
   const [newLessonsArr, setNewLessonsArr] = useState<Lesson[]>([]);
   const [popularLessonsArr, setPopularLessonsArr] = useState<Lesson[]>([]);
   const [categoryName, setCategoryName] = useState<string>("");
   const [categoriesNames, setCategoriesNames] = useState<Category[]>();
-  console.log(categoriesNames);
+  const [rating, setRating]= useState<undefined|number>();
+  const [isAuthorized, setIsAuthorized]= useState(false);
+  const [isRatingFreezen, setIsRatingFreezen]= useState(false);
+  // Написат начальное значение реальное
```
Comment (Terapool) [inline src/pages/VideoViewPage.tsx:49]: Remove this comment, please

File: src/pages/VideoViewPage.tsx [hunk 8]
```diff
@@ -301,3 +371,7 @@ export const VideoViewPage = () => {
     </div>
   );
 };
+const mapStateToProps = (state: RootState) => {
+  return { user: state?.login?.user };
+};
```
Comment (Terapool) [inline src/pages/VideoViewPage.tsx:376]: emty line, then export default then empty line...

File: src/components/VideoRating.tsx [hunk 9]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
...
+  isRatingFreezen: boolean;
 }
-const VideoRating = ({ ratingProp, votesCount }: VideoRatingProps) => {
+const VideoRating = ({
+  ratingProp,
+  votesCount,
+  onGetNewRating,
+  isAuthorized,
+  isRatingFreezen,
+}: VideoRatingProps) => {
```
Comment (Terapool) [inline src/components/VideoRating.tsx:19]: Please, create component RatingStar and use it in VideoRating component to render star

File: src/components/VideoRating.tsx [hunk 10]
```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
...
         const ratingUnit: number = index + 1;
+
         return (
           <button
+            style={isAuthorized && !isRatingFreezen? { cursor: "pointer" } : { cursor: "auto" }}
             type="button"
             key={ratingUnit}
             className={ratingUnit <= (hover || rating) ? "on" : "off"}
-            onClick={() => setRating(ratingUnit)}
+            onClick={isAuthorized && !isRatingFreezen? () => {
```
Comment (Terapool) [inline src/components/VideoRating.tsx:34]: Too long, too many lines - please write onClick handler separately and then insert its name

**Review by Terapool — CHANGES_REQUESTED** [review]

**Review by Terapool — APPROVED** [review]

**Review by valer-sky — APPROVED** [review]

**Pull request closed without merging**